lib/module.js

```diff
@@ -516,8 +517,12 @@ Module.prototype.load = function(filename) {
   this.filename = filename;
   this.paths = Module._nodeModulePaths(path.dirname(filename));
 
-  var extension = path.extname(filename) || '.js';
-  if (!Module._extensions[extension]) extension = '.js';
+  var extension = path.extname(filename);
+  if (!extension || !Module._extensions[extension]) {
+    internalUtil.deprecate(() => {
```

Comment:
Since this is re-wrapping a fresh function each time, will this log a deprecation warning on each require?

Reply:
oh gosh, yes we should move that out.